content/2017.03.building-projects-as-a-way-to-learn.rst

```diff
@@ -0,0 +1,222 @@
+
+- Firefox Hello: Building a WebRTC signaling service.
+- MSISDN auth: Building a password less BrowserID authentication
+  service using a phone number.
+- Syncto: Browsing Firefox Sync using the Kinto protocol.
+- Kinto: Building synchronization services with backup and sharing
+  capabilities in mind.
+- RemoteServiceWorker: Building a Firefox Service Worker that does
+  heavy calculation in the cloud, to save battery life on mobile.
+- WebPush channels: Building a PubSub on top of WebPush
+
+While most of these projects have been built, deployed to production
+and eventually decommissioned, some of them are still actively used by
+Firefox users everyday.
+
+
+How do you keep motivated building things that are thrown away?
+===============================================================
+
+Let's be honest: everything we build — cars, houses, and even software —
+is meant to be eventually thrown away. That's why recycling is so
+important.
+
+The question is why do we keep building them nevertheless.
+
+Our daily job is to help Mozilla make the web a safe place for users
+and prevent their privacy from being compromised.
+
+One way of doing so is to build software that empowers people to use
+the Internet while being in charge of their own data.
+
+When users give their data to big companies, they know they are giving
+them to much power. In the meantime these companies are offering a
+service that solve people problems, and they get addicted to it and
+come back for more.
+
+We alone cannot compete with companies that are spending millions of
+dollars building great products to capture users.
+
+However, we can build software that, on the one hand, is useful for
+the Mozilla mission and enable us to fix Mozilla issues quickly while
+building Firefox and, on the other hand, empowers developers to build
+great software where users choose where they want to store and control
+their data.
+
+What matters rather than building software is the vision that this
+software carries.
+
+If you don't have a vision, what are you building? Sure, writing lines
+of code is cheap and fast. You can build a prototype in two weeks and
+iterate until you are happy with the Minimum Viable Product.
+
+Note that, it might not be so cheap anymore when you start adding
+documentation and maintenance costs.
+
+The product might be used, and might get some traction for months or
+even years but it will be eventually forgotten in a git repository
+somewhere forever.
+
+Writing software to solve problems is not enough, to keep motivated on
+the long term. You need a greater goal than just writing code.
+
+Knowing that, how do you keep motivated building things that are
+thrown away?
+
+Different people or communities might have different answers.
+
+For us the answer is: As well as solving the problem, what can we
+learn while building this solution that can be useful for us and for
+our community to help us go faster and build more robust software
+later.
```

Comment:
This is framed as a question, but the previous paragraph mentions `answers`. I also think this paragraph is a little unwieldy. How about `As well as solving the problem, let's learn what we can ... our community. This way, we can ...`